src/renderers/webgl/WebGLState.js

```diff
@@ -337,168 +328,87 @@ THREE.WebGLState = function ( gl, extensions, paramThreeToGL ) {
-						break;
-
-					case THREE.EqualDepth:
-
-						gl.depthFunc( gl.EQUAL );
-						break;
-
-					case THREE.GreaterEqualDepth:
-
-						gl.depthFunc( gl.GEQUAL );
-						break;
-
-					case THREE.GreaterDepth:
-
-						gl.depthFunc( gl.GREATER );
-						break;
-
-					case THREE.NotEqualDepth:
+	this.enableDepthBuffer = function () {
 
-						gl.depthFunc( gl.NOTEQUAL );
-						break;
-
-					default:
+		this.depthBuffer.enabled = true;
 
-						gl.depthFunc( gl.LEQUAL );
-
-				}
+	};
 
-			} else {
+	this.disableDepthBuffer = function () {
 
-				gl.depthFunc( gl.LEQUAL );
+		this.depthBuffer.enabled = false;
 
-			}
+	};
 
-			currentDepthFunc = depthFunc;
+	this.setDepthFunc = function ( depthFunc ) {
 
-		}
+		this.depthBuffer.func( depthFunc );
 
 	};
 
 	this.setDepthTest = function ( depthTest ) {
 
-		if ( depthTest ) {
-
-			this.enable( gl.DEPTH_TEST );
-
-		} else {
-
-			this.disable( gl.DEPTH_TEST );
-
-		}
+		this.depthBuffer.test( depthTest );
 
 	};
 
 	this.setDepthWrite = function ( depthWrite ) {
 
-		// TODO: Rename to setDepthMask
+		this.depthBuffer.mask( depthWrite );
 
-		if ( currentDepthWrite !== depthWrite ) {
+	};
 
-			gl.depthMask( depthWrite );
-			currentDepthWrite = depthWrite;
+	this.enableColorBuffer = function () {
 
-		}
+		this.colorBuffer.enabled = true;
 
 	};
 
-	this.setColorWrite = function ( colorWrite ) {
+	this.disableColorBuffer = function () {
 
-		// TODO: Rename to setColorMask
+		this.colorBuffer.enabled = false;
 
-		if ( currentColorWrite !== colorWrite ) {
+	};
 
-			gl.colorMask( colorWrite, colorWrite, colorWrite, colorWrite );
-			currentColorWrite = colorWrite;
+	this.setColorWrite = function ( colorWrite ) {
 
-		}
+		this.colorBuffer.mask( colorWrite );
 
 	};
 
 	this.setStencilFunc = function ( stencilFunc, stencilRef, stencilMask ) {
 
-		if ( currentStencilFunc !== stencilFunc ||
-				 currentStencilRef 	!== stencilRef 	||
-				 currentStencilMask !== stencilMask ) {
-
-			gl.stencilFunc( stencilFunc,  stencilRef, stencilMask );
-
-			currentStencilFunc = stencilFunc;
-			currentStencilRef  = stencilRef;
-			currentStencilMask = stencilMask;
-
-		}
+		this.stencilBuffer.func( stencilFunc, stencilRef, stencilMask );
 
 	};
 
 	this.setStencilOp = function ( stencilFail, stencilZFail, stencilZPass ) {
 
-		if ( currentStencilFail	 !== stencilFail 	||
-				 currentStencilZFail !== stencilZFail ||
-				 currentStencilZPass !== stencilZPass ) {
-
-			gl.stencilOp( stencilFail,  stencilZFail, stencilZPass );
-
-			currentStencilFail  = stencilFail;
-			currentStencilZFail = stencilZFail;
-			currentStencilZPass = stencilZPass;
-
-		}
+		this.stencilBuffer.op( stencilFail, stencilZFail, stencilZPass );
 
 	};
 
 	this.setStencilTest = function ( stencilTest ) {
 
-		if ( stencilTest ) {
+		this.stencilBuffer.test( stencilTest );
 
-			this.enable( gl.STENCIL_TEST );
-
-		} else {
+	};
 
-			this.disable( gl.STENCIL_TEST );
+	this.setStencilWrite = function ( stencilWrite ) {
```

Comment:
Yes, unfortunately we already have these defined in previous releases.

https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLState.js#L465-L476

We're trying to move away from these names. The diff may be a bit confusing at the moment though.